reposerver/repository/repository.go

```diff
@@ -1719,13 +1772,35 @@ func (s *Service) newHelmClientResolveRevision(repo *v1alpha1.Repository, revisi
 	return helmClient, version.String(), nil
 }
 
+// directoryPermissionInitializer ensures the directory has read/write/execute permissions and returns
+// a function that can be used to remove all permissions.
+func directoryPermissionInitializer(rootPath string) goio.Closer {
+	if _, err := os.Stat(rootPath); err == nil {
```

Comment:
If `rootPath` does not exist `os.Stat` will return an error of type `os.ErrNotExist`. Is this something we should care about in this function?

Reply:
If the directory is missing then there is no need to restore permissions and the function should just exit. So I think we are fine.